{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "identified_table = pd.read_csv('test/data/identified/EMX1_identifiedOfftargets.txt', sep='\\t', index_col=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#BED Chromosome</th>\n",
       "      <th>BED Min.Position</th>\n",
       "      <th>BED Max.Position</th>\n",
       "      <th>BED Name</th>\n",
       "      <th>Filename</th>\n",
       "      <th>WindowIndex</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>+.mi</th>\n",
       "      <th>...</th>\n",
       "      <th>Length</th>\n",
       "      <th>BED off-target Chromosome</th>\n",
       "      <th>BED off-target start</th>\n",
       "      <th>BED off-target end</th>\n",
       "      <th>BED off-target name</th>\n",
       "      <th>BED Score</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Cells</th>\n",
       "      <th>Targetsite</th>\n",
       "      <th>Target Sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr15:44108746-44110769</td>\n",
       "      <td>1007</td>\n",
       "      <td>1025</td>\n",
       "      <td>chr15:44108746-44110769_1017_189</td>\n",
       "      <td>EMX1.sam</td>\n",
       "      <td>1</td>\n",
       "      <td>15:44108746-44110769</td>\n",
       "      <td>1017</td>\n",
       "      <td>GTAGACAAGAGTCTAAGCAGAAGAAGAAGAGAGCCACTACCCAACC...</td>\n",
       "      <td>116</td>\n",
       "      <td>...</td>\n",
       "      <td>23.0</td>\n",
       "      <td>chr15:44108746-44110769</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>1023.0</td>\n",
       "      <td>chr15:44108746-44110769_1017_189</td>\n",
       "      <td>1.0</td>\n",
       "      <td>+</td>\n",
       "      <td>EMX_site1</td>\n",
       "      <td>EMX1</td>\n",
       "      <td>GAGTCCGAGCAGAAGAAGAANGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1:236259170-236261754</td>\n",
       "      <td>1465</td>\n",
       "      <td>1486</td>\n",
       "      <td>chr1:236259170-236261754_1486_7</td>\n",
       "      <td>EMX1.sam</td>\n",
       "      <td>8</td>\n",
       "      <td>1:236259170-236261754</td>\n",
       "      <td>1486</td>\n",
       "      <td>ATGGAGCAGGCGACCAGGGGTGACTCAGAATGGAGCAGGTGACCAG...</td>\n",
       "      <td>7</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>EMX_site1</td>\n",
       "      <td>EMX1</td>\n",
       "      <td>GAGTCCGAGCAGAAGAAGAANGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1:236259170-236261754</td>\n",
       "      <td>1531</td>\n",
       "      <td>1539</td>\n",
       "      <td>chr1:236259170-236261754_1531_5</td>\n",
       "      <td>EMX1.sam</td>\n",
       "      <td>10</td>\n",
       "      <td>1:236259170-236261754</td>\n",
       "      <td>1531</td>\n",
       "      <td>GGGGTGACTCAGAATGGAGCAGGTGACCAGGGGAATAGACGTTAAC...</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>EMX_site1</td>\n",
       "      <td>EMX1</td>\n",
       "      <td>GAGTCCGAGCAGAAGAAGAANGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr2:73159981-73162004</td>\n",
       "      <td>1008</td>\n",
       "      <td>1024</td>\n",
       "      <td>chr2:73159981-73162004_1017_489</td>\n",
       "      <td>EMX1.sam</td>\n",
       "      <td>12</td>\n",
       "      <td>2:73159981-73162004</td>\n",
       "      <td>1017</td>\n",
       "      <td>AAGGGCCTGAGTCCGAGCAGAAGAAGAAGGGCTCCCATCACATCAA...</td>\n",
       "      <td>243</td>\n",
       "      <td>...</td>\n",
       "      <td>23.0</td>\n",
       "      <td>chr2:73159981-73162004</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>1023.0</td>\n",
       "      <td>chr2:73159981-73162004_1017_489</td>\n",
       "      <td>1.0</td>\n",
       "      <td>+</td>\n",
       "      <td>EMX_site1</td>\n",
       "      <td>EMX1</td>\n",
       "      <td>GAGTCCGAGCAGAAGAAGAANGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr3:197899267-197901348</td>\n",
       "      <td>1075</td>\n",
       "      <td>1081</td>\n",
       "      <td>chr3:197899267-197901348_1080_10</td>\n",
       "      <td>EMX1.sam</td>\n",
       "      <td>16</td>\n",
       "      <td>3:197899267-197901348</td>\n",
       "      <td>1080</td>\n",
       "      <td>TTAGGGTTAGGGTTAGGGTTAGGGTTCGGGTTTAGGGTTCAGGTTT...</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>EMX_site1</td>\n",
       "      <td>EMX1</td>\n",
       "      <td>GAGTCCGAGCAGAAGAAGAANGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr6:9117792-9119815</td>\n",
       "      <td>1007</td>\n",
       "      <td>1007</td>\n",
       "      <td>chr6:9117792-9119815_1007_4</td>\n",
       "      <td>EMX1.sam</td>\n",
       "      <td>18</td>\n",
       "      <td>6:9117792-9119815</td>\n",
       "      <td>1007</td>\n",
       "      <td>ATGTCCTCAGAGTTCTGTCCATTCTTCTTCTGCTCAGACGTTTTGT...</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>23.0</td>\n",
       "      <td>chr6:9117792-9119815</td>\n",
       "      <td>1000.0</td>\n",
       "      <td>1023.0</td>\n",
       "      <td>chr6:9117792-9119815_1007_4</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>EMX_site1</td>\n",
       "      <td>EMX1</td>\n",
       "      <td>GAGTCCGAGCAGAAGAAGAANGG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6 rows Ã— 33 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            #BED Chromosome  BED Min.Position  BED Max.Position  \\\n",
       "0   chr15:44108746-44110769              1007              1025   \n",
       "1  chr1:236259170-236261754              1465              1486   \n",
       "2  chr1:236259170-236261754              1531              1539   \n",
       "3    chr2:73159981-73162004              1008              1024   \n",
       "4  chr3:197899267-197901348              1075              1081   \n",
       "5      chr6:9117792-9119815              1007              1007   \n",
       "\n",
       "                           BED Name  Filename  WindowIndex  \\\n",
       "0  chr15:44108746-44110769_1017_189  EMX1.sam            1   \n",
       "1   chr1:236259170-236261754_1486_7  EMX1.sam            8   \n",
       "2   chr1:236259170-236261754_1531_5  EMX1.sam           10   \n",
       "3   chr2:73159981-73162004_1017_489  EMX1.sam           12   \n",
       "4  chr3:197899267-197901348_1080_10  EMX1.sam           16   \n",
       "5       chr6:9117792-9119815_1007_4  EMX1.sam           18   \n",
       "\n",
       "              Chromosome  Position  \\\n",
       "0   15:44108746-44110769      1017   \n",
       "1  1:236259170-236261754      1486   \n",
       "2  1:236259170-236261754      1531   \n",
       "3    2:73159981-73162004      1017   \n",
       "4  3:197899267-197901348      1080   \n",
       "5      6:9117792-9119815      1007   \n",
       "\n",
       "                                            Sequence  +.mi  \\\n",
       "0  GTAGACAAGAGTCTAAGCAGAAGAAGAAGAGAGCCACTACCCAACC...   116   \n",
       "1  ATGGAGCAGGCGACCAGGGGTGACTCAGAATGGAGCAGGTGACCAG...     7   \n",
       "2  GGGGTGACTCAGAATGGAGCAGGTGACCAGGGGAATAGACGTTAAC...     0   \n",
       "3  AAGGGCCTGAGTCCGAGCAGAAGAAGAAGGGCTCCCATCACATCAA...   243   \n",
       "4  TTAGGGTTAGGGTTAGGGTTAGGGTTCGGGTTTAGGGTTCAGGTTT...     0   \n",
       "5  ATGTCCTCAGAGTTCTGTCCATTCTTCTTCTGCTCAGACGTTTTGT...     1   \n",
       "\n",
       "            ...             Length  BED off-target Chromosome  \\\n",
       "0           ...               23.0    chr15:44108746-44110769   \n",
       "1           ...                NaN                        NaN   \n",
       "2           ...                NaN                        NaN   \n",
       "3           ...               23.0     chr2:73159981-73162004   \n",
       "4           ...                NaN                        NaN   \n",
       "5           ...               23.0       chr6:9117792-9119815   \n",
       "\n",
       "   BED off-target start  BED off-target end               BED off-target name  \\\n",
       "0                1000.0              1023.0  chr15:44108746-44110769_1017_189   \n",
       "1                   NaN                 NaN                               NaN   \n",
       "2                   NaN                 NaN                               NaN   \n",
       "3                1000.0              1023.0   chr2:73159981-73162004_1017_489   \n",
       "4                   NaN                 NaN                               NaN   \n",
       "5                1000.0              1023.0       chr6:9117792-9119815_1007_4   \n",
       "\n",
       "   BED Score  Strand      Cells  Targetsite          Target Sequence  \n",
       "0        1.0       +  EMX_site1        EMX1  GAGTCCGAGCAGAAGAAGAANGG  \n",
       "1        NaN     NaN  EMX_site1        EMX1  GAGTCCGAGCAGAAGAAGAANGG  \n",
       "2        NaN     NaN  EMX_site1        EMX1  GAGTCCGAGCAGAAGAAGAANGG  \n",
       "3        1.0       +  EMX_site1        EMX1  GAGTCCGAGCAGAAGAAGAANGG  \n",
       "4        NaN     NaN  EMX_site1        EMX1  GAGTCCGAGCAGAAGAAGAANGG  \n",
       "5        1.0       -  EMX_site1        EMX1  GAGTCCGAGCAGAAGAAGAANGG  \n",
       "\n",
       "[6 rows x 33 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "identified_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "on_targets = identified_table[identified_table['Mismatches'] == 0]\n",
    "off_targets = identified_table[identified_table['Mismatches'] > 0]\n",
    "cleavage_sites = pd.concat([on_targets, off_targets])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr2:73159981-73162004\n",
      "chr15:44108746-44110769\n",
      "chr6:9117792-9119815\n"
     ]
    }
   ],
   "source": [
    "for i, site in cleavage_sites.iterrows():\n",
    "    print(site['#BED Chromosome'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>10</th>\n",
       "      <th>11</th>\n",
       "      <th>12</th>\n",
       "      <th>13</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr15</td>\n",
       "      <td>44108746</td>\n",
       "      <td>44110769</td>\n",
       "      <td>chr15:44108746-44110769_1017_189</td>\n",
       "      <td>chr15</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>44096690</td>\n",
       "      <td>44117000</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000140259.6\"; transcript_id \"EN...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr15</td>\n",
       "      <td>44108746</td>\n",
       "      <td>44110769</td>\n",
       "      <td>chr15:44108746-44110769_1017_189</td>\n",
       "      <td>chr15</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>transcript</td>\n",
       "      <td>44096690</td>\n",
       "      <td>44117000</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000140259.6\"; transcript_id \"EN...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr15</td>\n",
       "      <td>44108746</td>\n",
       "      <td>44110769</td>\n",
       "      <td>chr15:44108746-44110769_1017_189</td>\n",
       "      <td>chr15</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>CDS</td>\n",
       "      <td>44109427</td>\n",
       "      <td>44109646</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>gene_id \"ENSG00000140259.6\"; transcript_id \"EN...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr15</td>\n",
       "      <td>44108746</td>\n",
       "      <td>44110769</td>\n",
       "      <td>chr15:44108746-44110769_1017_189</td>\n",
       "      <td>chr15</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>exon</td>\n",
       "      <td>44109427</td>\n",
       "      <td>44109646</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000140259.6\"; transcript_id \"EN...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr2</td>\n",
       "      <td>73159981</td>\n",
       "      <td>73162004</td>\n",
       "      <td>chr2:73159981-73162004_1017_489</td>\n",
       "      <td>chr2</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>73143389</td>\n",
       "      <td>73162020</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000135638.9\"; transcript_id \"EN...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      0         1         2                                 3      4       5   \\\n",
       "0  chr15  44108746  44110769  chr15:44108746-44110769_1017_189  chr15  HAVANA   \n",
       "1  chr15  44108746  44110769  chr15:44108746-44110769_1017_189  chr15  HAVANA   \n",
       "2  chr15  44108746  44110769  chr15:44108746-44110769_1017_189  chr15  HAVANA   \n",
       "3  chr15  44108746  44110769  chr15:44108746-44110769_1017_189  chr15  HAVANA   \n",
       "4   chr2  73159981  73162004   chr2:73159981-73162004_1017_489   chr2  HAVANA   \n",
       "\n",
       "           6         7         8  9  10 11  \\\n",
       "0        gene  44096690  44117000  .  -  .   \n",
       "1  transcript  44096690  44117000  .  -  .   \n",
       "2         CDS  44109427  44109646  .  -  2   \n",
       "3        exon  44109427  44109646  .  -  .   \n",
       "4        gene  73143389  73162020  .  +  .   \n",
       "\n",
       "                                                  12  13  \n",
       "0  gene_id \"ENSG00000140259.6\"; transcript_id \"EN...   0  \n",
       "1  gene_id \"ENSG00000140259.6\"; transcript_id \"EN...   0  \n",
       "2  gene_id \"ENSG00000140259.6\"; transcript_id \"EN...   0  \n",
       "3  gene_id \"ENSG00000140259.6\"; transcript_id \"EN...   0  \n",
       "4  gene_id \"ENSG00000135638.9\"; transcript_id \"EN...   0  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotations = pd.read_csv('test/reporting/data/tmp/EMX1-10nm_annotation.bed',sep='\\t', index_col=False, header=None)\n",
    "annotations.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "a = 'chr15:44108746-44110769_1017_189'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>10</th>\n",
       "      <th>11</th>\n",
       "      <th>12</th>\n",
       "      <th>13</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr15</td>\n",
       "      <td>44108746</td>\n",
       "      <td>44110769</td>\n",
       "      <td>chr15:44108746-44110769_1017_189</td>\n",
       "      <td>chr15</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>44096690</td>\n",
       "      <td>44117000</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000140259.6\"; transcript_id \"EN...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      0         1         2                                 3      4       5   \\\n",
       "0  chr15  44108746  44110769  chr15:44108746-44110769_1017_189  chr15  HAVANA   \n",
       "\n",
       "     6         7         8  9  10 11  \\\n",
       "0  gene  44096690  44117000  .  -  .   \n",
       "\n",
       "                                                  12  13  \n",
       "0  gene_id \"ENSG00000140259.6\"; transcript_id \"EN...   0  "
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "b = annotations[3] == a\n",
    "c = annotations[6] == 'gene'\n",
    "d = annotations[b & c]\n",
    "d.head(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "44108746"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d.get_value(0,1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [Root]",
   "language": "python",
   "name": "Python [Root]"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
